src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -2012,5 +2013,35 @@ public String getVariableNamePattern() {
 					}
 				})
 				.serializer(new EnumSerializer<>(Attribute.class)));
+		EnumUtils<PlayerFishEvent.State> fishStateUtils = new EnumUtils<>(PlayerFishEvent.State.class, "fishing states");
+		Classes.registerClass(new ClassInfo<>(PlayerFishEvent.State.class, "fishingstate")
+				.user("fish(ing|) ?states?")
+				.name("Fishing state")
+				.description("Represents the fishing state in a <a href='events.html#fishing'>fishing</a> event")
+				.usage(fishStateUtils.getAllNames())
+				.since("INSERT VERSION")
+				.parser(new Parser<PlayerFishEvent.State>() {
+					@Nullable
+					@Override
+					public PlayerFishEvent.State parse(String s, ParseContext context) {
+						return fishStateUtils.parse(s);
+					}
+					
+					@Override
+					public String toString(PlayerFishEvent.State o, int flags) {
+						return fishStateUtils.toString(o, flags);
+					}
+					
+					@Override
+					public String toVariableNameString(PlayerFishEvent.State o) {
+						return o.name();
+					}
+					
+					@Override
+					public String getVariableNamePattern() {
+						return "\\S";
```

Comment:
this is not right \s means whitespace. you would probably just want to use .+ here as it could technically be just about anything since aliases can be changed

Reply:
\s means whitespace, \S everything but whitespace